### DISCUSSION TITLE
X1850 + SC6000 + Hak5 Packet Squirrel 🐿 == Fun :)

### DISCUSSION THREAD

**Discussion by chrisle** [MarByteBeep/StageLinq#16] [Show and tell]
Hello everyone!

**Open invitation to this community**: I will capture PCAP files and put them in [this repo](https://github.com/chrisle/stagelinq-pcap)

- (1) 🐿 Hak5 [Packet Squirrel](https://docs.hak5.org/packet-squirrel/) used to capture packets to a USB stick 
- (1) X1850 Mixer (firmware v1.6)
- (2) SC6000 Players (firmware 2.3.1)
- (1) Denon Prime Go
- Beatport Link login
- Dropbox login

I forked this repo, add some additional functionality to it, and send PRs.

@MarByteBeep Thank you BTW for this library!

**Comment by honusz** [comment]
Great idea! It's good to have an organized place for stuff, I'm forever recapturing pcaps.

Just the other day I started doing some captures between an x1800 and sc6000 (with a port-mirroring switch).
Goal is to better understand the on-board Engine Lighting.

**Reply by chrisle** [reply]
@honusz Awesome!

I have a few immediate goals for myself:

- Create proper documentation like Dysentery did for ProDJ-Link.
- Figure out why sometimes (seemingly at random) the devices stop allowing them to be discovered.
- Figure out how the hardware knows which one is "sync lead" (critical for my app)

Future goals:

- Understand the BeatInfo work you've done
- Figure out what the "Syncing" service is doing
- Figure out how Beatport Link really works
- Use Engine Lighting & SoundSwitch to DMX that would actually control OBS via websocket 🤪
- Fully setup DMX light show for myself with SoundSwitch & custom code :)

**Reply by honusz** [reply]
@chrisle Cool! Definitely a lot of crossover in our interests. 

> Figure out how the hardware knows which one is "sync lead" (critical for my app)

Yeah, this is something that interests me as well. Unlike all the other related functions (which player is 'On-air', which player is Engine Lighting taking cues from), the players manage to determine Sync/Lead without any data from the mixer.

I've noticed the following behaviours with it:
Let's say you start playing Track A, then start Track B four bars later. Track A remains 'Lead' until either:
* Track A stops playing, either from reaching the end or from pressing stop.
* Loop is engaged on Track 
* The ~Master~ Lead button is pressed on Track B's player.

There may be some other conditions that cause a swap, but I haven't come across them yet.

**Comment by chrisle** [comment]
Other random things I've learned last night:

- It hits http://ipv4.connman.net/online/status.html to figure out if it's online or not.
- It checks autoupdate.airmusictech.com for updates
- Before you turn it off it sends telemetry to sentry.io hah
